data/main/src/main/kotlin/nl/q42/template/data/main/UserRepositoryImpl.kt

```diff
@@ -22,13 +22,19 @@ internal class UserRepositoryImpl @Inject constructor(
         // get remotely
         val userEntityActionResult = userRemoteDataSource.getUser()
         // store locally
-        userEntityActionResult.getDataOrNull()?.let { userEntity ->
-            userLocalDataSource.setUser(userEntity)
-        }
+        when (userEntityActionResult) {
+            is ActionResult.Success -> {
+                userLocalDataSource.setUser(userEntityActionResult.data)
+            }
 
+            is ActionResult.Error -> {
```

Comment:
We can move this error handler one level deeper, to the RemoteDataSource. Handling errors deeper /earlieris better, so we should probably do that, right?

Reply:
the success needs to be done here, the logging can indeed go down.